Miracast/MiracastPlayer/MiracastGstPlayer.cpp

```diff
@@ -0,0 +1,932 @@
+
+        if (!player_stats_monitor_delay)
+        {
+            MIRACASTLOG_TRACE("Statistics Disabled as '/opt/miracast_player_stats' file not exists");
+        }
+        else{
+            std::string word;
+            player_stats_monitor_delay >> word;
+            player_stats_monitor_delay.close();
+
+            if (word.empty())
+            {
+                isInteger = false;
+            }
+            else
+            {
+                for (char c : word) {
+                    if (!isdigit(c)) {
+                        isInteger = false;
+                        break;
+                    }
+                }
+            }
+
+            if (isInteger && (time_interval_sec = std::atoi(word.c_str())))
+            {
+                self->get_player_statistics();
+            }
+            else
+            {
+                MIRACASTLOG_ERROR("The content of the file is not an integer\n");
+            }
+        }
+        sleep(time_interval_sec);
+    }
+    self->m_player_statistics_tid = 0;
+    MIRACASTLOG_TRACE("Exiting..!!!");
+    pthread_exit(nullptr);
+}
+
+double MiracastGstPlayer::getDuration( GstElement *pipeline )
+{
+    MIRACASTLOG_TRACE("Entering..!!!");
+    gint64 duration = 0;
+    double ret = 0.0f;
+
+    if ( nullptr == pipeline )
+    {
+        pipeline = m_pipeline;
+    }
+
+    if (gst_element_query_duration(pipeline, GST_FORMAT_TIME, &duration) && GST_CLOCK_TIME_IS_VALID(duration))
+    {
+        ret = static_cast<double>(duration) / GST_SECOND;
+    }
+    MIRACASTLOG_TRACE("Exiting..!!!");
+    return ret;
+}
+
+double MiracastGstPlayer::getCurrentPosition(GstElement *pipeline)
+{
+    MIRACASTLOG_TRACE("Entering..!!!");
+    gint64 position = 0;
+
+    if ( nullptr == pipeline )
+    {
+        pipeline = m_pipeline;
+    }
+
+    if (gst_element_query_position(pipeline, GST_FORMAT_TIME, &position))
+    {
+        position = static_cast<double>(position) / GST_SECOND;
+    }
+    MIRACASTLOG_TRACE("Exiting..!!!");
+    return position;
+}
+
+bool MiracastGstPlayer::seekTo(double seconds, GstElement *pipeline )
+{
+    MIRACASTLOG_TRACE("Entering..!!!");
+    bool ret = false;
+    gint64 cur = static_cast<gint64>(seconds * GST_SECOND);
+    m_currentPosition = seconds;
+
+    MIRACASTLOG_VERBOSE("seekToPos=%f", seconds);
+
+    if ( nullptr == pipeline )
+    {
+        pipeline = m_pipeline;
+    }
+
+    if (!gst_element_seek(pipeline, 1.0, GST_FORMAT_TIME, GST_SEEK_FLAG_FLUSH, GST_SEEK_TYPE_SET, cur, GST_SEEK_TYPE_NONE, GST_CLOCK_TIME_NONE))
+    {
+        MIRACASTLOG_VERBOSE("SeekToPos success\n");
+        ret = true;
+    }
+    else
+    {
+        MIRACASTLOG_VERBOSE("seekToPos failed");
+    }
+    MIRACASTLOG_TRACE("Exiting..!!!");
+    return ret;
+}
+
+void MiracastGstPlayer::print_pipeline_state(GstElement *pipeline)
+{
+    MIRACASTLOG_TRACE("Entering..!!!");
+
+    if ( nullptr == pipeline )
+    {
+        pipeline = m_pipeline;
+    }
+
+    if ( nullptr == pipeline )
```

Comment:
can be ombine with previous if

Reply:
Addressed in https://github.com/rdkcentral/rdkservices/pull/4618/commits/abb5db62d2585d5e30a40d8f7c1645336732c0ea